luxjs/forms/utils.js

```diff
@@ -42,26 +61,90 @@ angular.module('lux.form.utils', ['lux.services'])
                     } else {
                         name = val[nameOpts.source];
                     }
+
+                    // If the value already exists in the select, don't
+                    // add it again.
+                    for (var i=0; i<options.length; i++) {
+                        if (options[i].id === val[id]) return;
+                    }
+
                     options.push({
                         id: val[id],
                         name: name
                     });
+
+                });
+                // Sort list alphabetically by name.
+                options.sort(function(a,b) {
+                    var one = a.name.toLowerCase();
```

Comment:
Adding those queries works on page load and lazy load but if you search when half the data has been lazy loaded then continue lazy loading, it leaves the search in the middle of the options rather than correctly alphabetised. I'm not sure if this sort function is the best / most efficient option but am very open to other suggestions!

<img width="240" alt="screen shot 2015-12-29 at 12 52 38" src="https://cloud.githubusercontent.com/assets/9137745/12034628/9d487c3c-ae2b-11e5-9a9f-f9b806b1e9e2.png">